### ISSUE TITLE
Container Service SDK doesn't return an error message for 400 errors

### ISSUE THREAD

**Issue by tombuildsstuff** [Azure/azure-sdk-for-go#2085]
👋 

We've noticed that [the Container Service SDK](https://github.com/Azure/azure-sdk-for-go/tree/v16.2.1/services/containerservice/mgmt/2017-09-30/containerservice) doesn't return a complete error message when an `err` object is returned (here's the relevant issue on our side, but we've seen this for other issues too](https://github.com/terraform-providers/terraform-provider-azurerm/issues/1416)):

```
containerservice.ManagedClustersClient#CreateOrUpdate: Failure sending request: StatusCode=400 -- Original Error: Code="" Message=""
```

The full error message is available in the HTTP Response - such that this just isn't being parsed out correctly; as such would it be possible to fix the SDK so that this error is raised?

Thanks!

**Comment by jhendrixMSFT** [comment]
@tombuildsstuff I don't suppose you have the response body handy, or even better a simple repro?  I've been having a hard time putting a repro together.  :(

**Comment by tombuildsstuff** [comment]
@jhendrixMSFT not to hand, in this case the error came from using an incorrect machine size - however I’ll try and put one together in the morning :)

**Comment by tombuildsstuff** [comment]
@jhendrixMSFT I've added a repro in this repo: https://github.com/tombuildsstuff/azure-sdk-for-go-bug-2085

**Comment by jhendrixMSFT** [comment]
Thanks a lot, will dig into this today.

**Comment by jhendrixMSFT** [comment]
The good news is this is fixed in recent builds.  I'll figure out which commit fixed it.

**Comment by jhendrixMSFT** [comment]
This was fixed in go-autorest v10.9.0 in the future state machine rewrite.  If possible I would recommend moving to v17 of the SDK in conjunction with v10.9 of go-autorest as the futures rewrite included some changes to our code generator which debuted in v17 of the SDK (this didn't introduce breaking changes for consumers, just the internals of some methods).
